app/services/katello/pulp/importer_comparison.rb

```diff
@@ -0,0 +1,31 @@
+module Katello
+  module Pulp
+    module ImporterComparison
+      def importer_matches?(generated_importer, capsule_importer)
+        if capsule_importer.try(:[], 'importer_type_id') != generated_importer.id
+          return false
+        end
+
+        generated_config = generated_importer.config
+        capsule_config = capsule_importer['config']
+        if generated_config['proxy_host'] == ""
+          generated_config.delete('proxy_host')
+          generated_config.delete('proxy_username')
+          generated_config.delete('proxy_password')
+          generated_config.delete('proxy_port')
+          capsule_config.delete('proxy_host')
+          capsule_config.delete('proxy_username')
+          capsule_config.delete('proxy_password')
+          capsule_config.delete('proxy_port')
+        end
+
+        if generated_config['proxy_password'] == "" && capsule_config['proxy_password'] == "*****"
+          generated_config.delete('proxy_password')
+          capsule_config.delete('proxy_password')
+        end
```

Comment:
to clarify, if `generated_config['proxy_password']` is set, it will always refresh the importer.  Is that what we want?

Reply:
Is your suggestion to _always_ ignore `proxy_password` in the comparison? Since pulp2 will always return that masked string (if set previously to any value), it seems like any comparison will fail.

Reply:
possibly, but i was more opening up debate :)  
I think this is the safter thing to do is the way you have it, although it makes https://github.com/Katello/katello/pull/8415 not really work if you're using an http proxy, although thats not really a problem introduced by this pr.  So i guess yes lets leave it the way you have it :+1:   but we should probably open a bug around this to discuss how to properly address it (especially with pulp3)

Reply:
Hopefully this _isn't_ an issue in pulp3.

Reply:
We could ask pulp2 to be fixed to resolve this. That's another option.

Reply:
I checked and pulp3 returns the full proxy url:
```
{"pulp_href":"/pulp/api/v3/remotes/file/file/c6688395-58f8-461a-bd3e-145083741543/","pulp_created":"2019-11-26T19:11:49.606085Z","name":"proxytest-137322","url":"http://foo.com/bar/PULP_MANIFEST","ca_cert":null,"client_cert":null,"client_key":null,"tls_validation":true,"proxy_url":"http://boobar:boobar@foo.com","pulp_last_updated":"2019-11-26T19:12:43.872594Z","download_concurrency":20,"policy":"immediate"}
```

however i suspect that might be a bug ;)

I think this is fine for now!